src/main/proto/ProgramService.proto

```diff
@@ -199,10 +200,46 @@ message JoinProgramRequest {
   google.protobuf.StringValue department = 5;
 }
 
+message Invitation {
+  google.protobuf.StringValue id = 1;
+}
+
 message JoinProgramResponse {
   User user = 1;
 }
 
 message RemoveProgramRequest {
   google.protobuf.StringValue program_short_name = 1;
 }
+
+message GetJoinProgramInviteRequest {
+  google.protobuf.StringValue invite_id = 1;
+}
+
+message JoinProgramInvite {
+  google.protobuf.StringValue id = 1;
+
+  google.protobuf.Timestamp created_at = 2;
+
+  google.protobuf.Timestamp expires_at = 3;
+
+  google.protobuf.Timestamp accepted_at = 4;
+
+  Program program = 5;
+
+  google.protobuf.StringValue user_email = 6;
```

Comment:
Should we replace the 4 user properties with a User message? (user_email, first_name, last_name, role)

Reply:
maybe not, it's harder to generate a mapper that convert database entity to grpc object if we embed User message here. I think it's better to keep in line with the underlying database table.

Reply:
But what would the consumer of this service want? They already have the concept of Users, and we are nesting a Program object, it would be logically consistent to have a User message. 

Get a second person to review the proto file so we get another opinion on this.

Reply:
Thanks for making this update :)